lib/workload_mapper_dsl.rb

```diff
@@ -32,6 +32,7 @@ def parameter(name, value)
   end
 
   def to_h
+    return nil unless service
```

Comment:
Wouldn't it be better to return an empty hash here, so that there are no checks needed when dealing with the results?

There also should be a unit test for a change like this as this is very fundamental API, which should be fully covered by tests.